packages/core/src/render3/util/discovery_utils.ts

```diff
@@ -181,18 +184,30 @@ export function getInjectionTokens(element: Element): any[] {
  * </my-app>
  * ```
  * Calling `getDirectives` on `<button>` will return an array with an instance of the `MyButton`
- * directive that is associated with the DOM element.
+ * directive that is associated with the DOM node.
  *
  * Calling `getDirectives` on `<my-comp>` will return an empty array.
  *
- * @param element DOM element for which to get the directives.
- * @returns Array of directives associated with the element.
+ * @param node DOM node for which to get the directives.
+ * @returns Array of directives associated with the node.
  *
  * @publicApi
  * @globalApi ng
  */
-export function getDirectives(element: Element): {}[] {
-  const context = loadLContext(element)!;
+export function getDirectives(node: Node): {}[] {
+  if (node instanceof Text) {
+    return [];
+  }
+  const context = loadLContext(node, false);
+
+  if (context === null) {
+    return [];
+  }
```

Comment:
May be add a quick note here?
```suggestion
  // Note: do not use `loadLContext` with `throwOnNotFound` argument defined as `true`,
  // as it might have performance implications in case throw happens. 
  const context = loadLContext(node, /* throwOnNotFound */ false);
  if (context === null) {
    return [];
  }
```